### ISSUE TITLE
Epoch timer improvement when using multi_scale=True

### ISSUE THREAD

**Issue by Tom-Forsyth** [ultralytics/ultralytics#20314]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Currently, when using `multi_scale=True`, the remaining time estimate of the current epoch seems to rapidly change. I am guessing this is because it is just using the current iterations per second, which changes every batch due to image size changing.

![Image](https://github.com/user-attachments/assets/854f8a95-6c29-4aea-85cb-0e54fc8c2bb0)

![Image](https://github.com/user-attachments/assets/eee39fe5-9c60-4d7a-b1e3-5efd7c0160de)

Instead, the elapsed time and iterations completed could be used, which would provide a stable estimate of the remaining time after a few batches.

### Use case

This will be useful for large datasets where the remaining time could fluctuate many minutes or even hours. Getting a good estimate of the actual time to train a model is important to plan how to allocate training resources.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Tom-Forsyth, thank you for your suggestion and for helping improve Ultralytics 🚀! We appreciate your feedback on the epoch timer behavior when using `multi_scale=True`.

This is an automated response to let you know that an Ultralytics engineer will assist you soon. In the meantime, we recommend reviewing our [Docs](https://docs.ultralytics.com/) for more information on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, as well as answers to common questions.

If you have any additional details, example logs, or screenshots to further illustrate your point, please feel free to add them. This helps us better understand your feature request and prioritize improvements.

Join the Ultralytics community where it suits you best:
- For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, check out [Discourse](https://community.ultralytics.com/)
- To share knowledge and ideas, join our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your input and for being a part of the Ultralytics community! 🏆

**Comment by glenn-jocher** [comment]
Hi @Tom-Forsyth, thank you for reporting this issue and for the detailed explanation. You're absolutely right - the timer fluctuation happens because when `multi_scale=True` is enabled, the image sizes change randomly between batches (from 0.5× to 1.5× of base image size), causing significant variations in processing time.

Your proposed solution of using elapsed time and iterations completed to calculate a running average is a good approach that would provide much more stable estimates, especially for large datasets.

Since you've indicated you're willing to submit a PR, we'd definitely welcome your contribution! The relevant code is in the trainer's progress reporting system. When implementing, consider calculating a moving average of processing times that's less sensitive to individual batch size variations.

Thanks for helping improve YOLO!

**Comment by Tom-Forsyth** [comment]
Taking a quick look, it seems that the progress bar is done using [tqdm](https://tqdm.github.io/docs/tqdm/)

There is a `smoothing` option that controls this behavior. `0.0` is the average speed, and `1.0` is the instantaneous speed. The default is `0.3`, which explains why it jumps around.

This should be as simple as adding a parameter to control the `smoothing` option.
